src/NodaTime.Test/Calendars/BclCalendars.cs

```diff
@@ -62,23 +62,18 @@ private static Calendar GetCalendar(string name)
         /// get day/month/year values to match without getting the calendar right, the calendar
         /// affects the day of week.
         /// </summary>
-        internal static CalendarSystem CalendarSystemForCalendar(Calendar bcl)
-        {
+        internal static CalendarSystem CalendarSystemForCalendar(Calendar bcl) =>
             // Yes, this is horrible... but the specific calendars aren't available to test
             // against in netstandard
-            switch (bcl.GetType().Name)
+            bcl.GetType().Name switch
             {
-                case "GregorianCalendar": return CalendarSystem.Iso;
-                case "HijriCalendar": return CalendarSystem.IslamicBcl;
-                case "HebrewCalendar": return CalendarSystem.HebrewCivil;
-                case "PersianCalendar": return bcl.IsLeapYear(1) ? CalendarSystem.PersianSimple : CalendarSystem.PersianAstronomical;
-                case "UmAlQuraCalendar": return CalendarSystem.UmAlQura;
-                case "JulianCalendar":
-                    return CalendarSystem.Julian;
-                default:
-                    // No idea - we can't test with this calendar...
-                    return null;
-            }
-        }
+                "GregorianCalendar" => CalendarSystem.Iso,
```

Comment:
Could you switch based on the actual class type now, instead of having to use `.GetType().Name()`, i.e.:

```cs
bcl switch
{
    GregorianCalendar _  => CalendarSystem.Iso,
    HijriCalendar _ => CalendarSystem.IslamicBcl,
//etc.
}
```

Not sure if the `_` is actually needed or can be left out altogether

Reply:
We couldn't do so before, not due to switch expressions but because we couldn't refer to the specific calendar system types in more limited environments.
Now we only support netstandard2.0, this might be feasible - thanks for highlighting it. Will have a look...